Library/Homebrew/api/formula.rb

```diff
@@ -15,29 +15,56 @@ def fetch(name)
           Homebrew::API.fetch "formula/#{name}.json"
         end
 
-        sig { returns(Hash) }
-        def all_formulae
-          @all_formulae ||= begin
-            json_formulae = Homebrew::API.fetch_json_api_file "formula.json",
-                                                              target: HOMEBREW_CACHE_API/"formula.json"
-
-            @all_aliases = {}
-            json_formulae.to_h do |json_formula|
-              json_formula["aliases"].each do |alias_name|
-                @all_aliases[alias_name] = json_formula["name"]
-              end
+        sig { returns(T::Boolean) }
+        def prepare_json_data
+          json_formulae, updated = Homebrew::API.fetch_json_api_file "formula.json",
+                                                                     target: HOMEBREW_CACHE_API/"formula.json"
 
-              [json_formula["name"], json_formula.except("name")]
+          @all_aliases = {}
+          @all_formulae = json_formulae.to_h do |json_formula|
+            json_formula["aliases"].each do |alias_name|
+              @all_aliases[alias_name] = json_formula["name"]
             end
+
+            [json_formula["name"], json_formula.except("name")]
+          end
+
+          updated
+        end
+        private :prepare_json_data
+
+        sig { returns(Hash) }
+        def all_formulae
+          unless defined?(@all_formulae)
+            json_updated = prepare_json_data
+            write_names_and_aliases(regenerate: json_updated)
           end
+
+          @all_formulae
```

Comment:
Same here, feels weird to set in another file. Not obvious from this function definition which method sets it.